kong/plugins/oauth2/migrations/001_14_to_15.lua

```diff
@@ -54,12 +54,24 @@ return {
 
       CREATE INDEX IF NOT EXISTS "oauth2_authorization_credential_id_idx" ON "oauth2_authorization_codes" ("credential_id");
       CREATE INDEX IF NOT EXISTS "oauth2_authorization_service_id_idx"    ON "oauth2_authorization_codes" ("service_id");
-      CREATE INDEX IF NOT EXISTS "oauth2_authorization_api_id_idx"        ON "oauth2_authorization_codes" ("api_id");
-
       CREATE INDEX IF NOT EXISTS "oauth2_tokens_credential_id_idx"        ON "oauth2_tokens" ("credential_id");
       CREATE INDEX IF NOT EXISTS "oauth2_tokens_service_id_idx"           ON "oauth2_tokens" ("service_id");
-      CREATE INDEX IF NOT EXISTS "oauth2_tokens_api_id_idx"               ON "oauth2_tokens" ("api_id");
 
+      DO $$
+      BEGIN
+        CREATE INDEX IF NOT EXISTS "oauth2_authorization_api_id_idx"
+                                ON "oauth2_authorization_codes" ("api_id");
+      EXCEPTION WHEN UNDEFINED_COLUMN THEN
+        -- Do nothing, accept existing state
```

Comment:
Adding this won't hurt, but I'm confused: this is `14_to_15`, which assumes a 0.14 state (where `api_id` exists) migrating into 0.15 (where `api_id` _still_ exists). How is it possible that this runs and `api_id` does not exist? Ahhh, we run all `core` migrations (bringing the core entities to 1.x) before running any `plugin` migrations, right? That explains it.

Err, that's a bit problematic. This means that every time we delete a column we have to go back to old migrations and fix them this way. One possible answer would be that core and migrations need to run migrations in lockstep going version-by-version, but if we consider a third party plugin that is installed _after_ Kong 1.x is installed, and has migrations to run, they do not have this luxury. _Every_ migration touching foreign fields would have to be written assuming that they might not be there in the future...

Ok, got it. (Thanks for reading my stream-of-consciousness rambling as I evolve my understanding of it. :) ) 

So let's do this — but something similar to this probably needs to be in every other plugin that uses `api_id` in migrations, right? A quick `grep` in `kong/plugins` shows lots of `api_id` in migrations. Perhaps that should be in a separate PR will all of them?